sandbox/cgroup/tracking.go

```diff
@@ -151,6 +152,24 @@ tryAgain:
 	return nil
 }
 
+// ConfirmSystemdAppTracking checks if systemd tracks this process as a snap app.
+//
+// If the application process is not tracked then ErrCannotTrackProcess is returned.
+func ConfirmSystemdAppTracking(securityTag string) error {
+	pid := osGetpid()
+	path, err := cgroupProcessPathInTrackingCgroup(pid)
+	if err != nil {
+		return err
+	}
+
+	regex := regexp.MustCompile(fmt.Sprintf("%s-.+\\.scope$", securityTag))
```

Comment:
MustCompile would panic in the event that securityTag produced invalid format. Perhaps a better check would be to do:

```
// the transient scope of the application carries the security tag, eg:
// snap.hello-world.sh-4706fe54-7802-4808-aa7e-ae8b567239e0.scope
if strings.HasPrefix(filepath.Base(path), securityTag + "-")) {
    ...
}
```

Reply:
great catch, thank you!

updated!